tests/components/dio_chacon/test_cover.py

```diff
@@ -0,0 +1,204 @@
+"""Test the Dio Chacon Cover sensor."""
+
+from collections.abc import Callable
+import logging
+from unittest.mock import AsyncMock
+
+from homeassistant.components.cover import (
+    ATTR_CURRENT_POSITION,
+    ATTR_POSITION,
+    DOMAIN as COVER_DOMAIN,
+    SERVICE_CLOSE_COVER,
+    SERVICE_OPEN_COVER,
+    SERVICE_SET_COVER_POSITION,
+    SERVICE_STOP_COVER,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+    CoverDeviceClass,
+    CoverEntityFeature,
+)
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_ENTITY_ID,
+    ATTR_FRIENDLY_NAME,
+    ATTR_SUPPORTED_FEATURES,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from tests.common import MockConfigEntry
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def test_cover(
```

Comment:
I would suggest that this test will be split up. Have 1 test that tests the entity properties, you can use snapshots for this, checkout `flexit_bacnet`. They use the `snapshot_platform` helper. If you then execute the tests with `pytest ./tests/components/dio_chacon --snapshot-update` it will generate a `.ambr` file with all the properties you are asserting here

Reply:
I cut in to separate actions from callbacks. There is no need to use snapshots between tests.